var/spack/repos/builtin/packages/cppunit/package.py

```diff
@@ -13,3 +13,31 @@ class Cppunit(AutotoolsPackage):
     url = "http://dev-www.libreoffice.org/src/cppunit-1.13.2.tar.gz"
 
     version('1.13.2', '0eaf8bb1dcf4d16b12bec30d0732370390d35e6f')
+
+    variant('cxxstd',
+            default='default',
+            values=('default', '98', '11', '14', '17'),
+            multi=False,
+            description='Use the specified C++ standard when building.')
+
+    def setup_environment(self, spack_env, run_env):
+        cxxstd = self.spec.variants['cxxstd'].value
+        cxxstdflag = ''
+        if cxxstd == '98':
+            cxxstdflag = self.compiler.cxx98_flag
+        elif cxxstd == '11':
+            cxxstdflag = self.compiler.cxx11_flag
+        elif cxxstd == '14':
+            cxxstdflag = self.compiler.cxx14_flag
+        elif cxxstd == '17':
+            cxxstdflag = self.compiler.cxx17_flag
+        elif cxxstd == 'default':
+            pass
+        else:
+            # The user has selected a (new?) legal value that we've
+            # forgotten to deal with here.
+            tty.die(
```

Comment:
This is supposed not to happen. The variant should issue an error during parsing if its value is not among the ones admitted, i.e. `('default', '98', '11', '14', '17')`.

Reply:
This is a consistency check for the case where the recipe author has added a value to the variant but has not taken account of it in the conditional.

Reply:
In that case it will be taken care by the line above with getattr, right?